### DISCUSSION TITLE
Add synchronous signing

### DISCUSSION THREAD

**Discussion by gfpacheco** [panva/jose#532] [Ideas]
Would it be too bad to add a `signSync` function?

In my case the JWT signing is just part of the auth configuration and I'd rather not have to make it async, that might be too hard to do on some serverless frameworks.

**Comment by panva** [comment]
It is not possible to have a non-Promise/callback based API. The module has to work in Browser and Web-like runtimes which are backed by [`Web Cryptography API`](https://w3c.github.io/webcrypto/) as the crypto runtime and that is an entirely Promise-based API.

No modern runtime has an issue with supporting async/await and even top-level await is ubiquitous.

**Reply by gfpacheco** [reply]
![Sadpepe](https://user-images.githubusercontent.com/3705660/234688150-76ba93f8-2475-4903-bcbf-7d73c1f636fe.png)

**Reply by AlfreddGco** [reply]
I'm needing this for retool since their transformers don't support asyn/await :(
